src/cryptography/hazmat/primitives/serialization/ssh.py

```diff
@@ -102,6 +104,18 @@ def _load_ssh_ecdsa_public_key(expected_key_type, decoded_data, backend):
     return ec.EllipticCurvePublicKey.from_encoded_point(curve, data)
 
 
+def _load_ssh_ed25519_public_key(expected_key_type, decoded_data, backend):
+    data, rest = _ssh_read_next_string(decoded_data)
+
+    if rest:
+        raise ValueError('Key body contains extra bytes.')
+
+    if len(data) != ed25519._ED25519_KEY_SIZE:
+        raise ValueError("Public key is not 32 bytes")
```

Comment:
`from_public_bytes` already raises this same exception, do we need this check?

Reply:
Probably not. Since it's an invariant of the key type and not a particular quirk of ssh it makes sense for us to just use the enforcement at the key loading layer.